config/configcompression/compressiontype.go

```diff
@@ -5,31 +5,34 @@ package configcompression // import "go.opentelemetry.io/collector/config/config
 
 import "fmt"
 
-type CompressionType string
+// CompressionType Deprecated [0.94.0]: use Type instead.
+type CompressionType = Type
+
+type Type string
```

Comment:
Should we also consider encapsulating this into a struct? /cc @mx-psi because there is an effort to do that for `component.Type`.

Reply:
In this case I don't think the struct is necessary since we don't need to enforce anything about the string itself. I believe the string alias is sufficient for enforcing the specific values during text unmarshaling.